File: microservices/competence-matching-service/src/main/java/com/github/miguelsombrero/osaan/competence_matching_service/domain/entity/Subscription.java
package com.github.miguelsombrero.osaan.competence_matching_service.domain.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.Instant;
import java.util.UUID;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class Subscription {
    private UUID id;
    private String userId;
    private String email;
    private String skill;
    private int rating;
    private Instant createdAt;
}

File: ui/osaan-ui/src/types/subscription.ts
import type { Rating } from './rating';

export interface Subscription {
  id: string;
  userId: string;
  email: string;
  skill: string;
  rating: Rating;
  createdAt: string;
}

export interface SubscriptionDraft {
  skill: string;
  rating: Rating;
}
